// survey-creator-angular/esm2015/tab-control/property-grid-placeholder-header.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
export class SidebarPropertyGridPlaceholderHeaderComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
SidebarPropertyGridPlaceholderHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarPropertyGridPlaceholderHeaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
SidebarPropertyGridPlaceholderHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: SidebarPropertyGridPlaceholderHeaderComponent, selector: "svc-side-bar-property-grid-placeholder-header", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div class=\"svc-side-bar__container-header svc-sidebar__header-container\">\n    <div class=\"svc-side-bar__container-title\">{{model.caption}}</div>\n  </div>\n</ng-template>", styles: [":host { display: none; }"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarPropertyGridPlaceholderHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-side-bar-property-grid-placeholder-header",
                    templateUrl: "./property-grid-placeholder-header.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-side-bar-property-grid-placeholder-header", SidebarPropertyGridPlaceholderHeaderComponent);
