// survey-creator-angular/esm2015/tab-control/tab-control.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular, AngularComponentFactory } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./tab-button.component";
import * as i2 from "./tabs.component";
export class TabControlComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
TabControlComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabControlComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
TabControlComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: TabControlComponent, selector: "svc-tab-control", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"model.sideBarClassName\">\n    <div class=\"svc-sidebar-tabs__top-container\">\n      <div class=\"svc-sidebar-tabs__collapse-button\">\n        <svc-tab-button [model]=\"model.expandCollapseAction\"></svc-tab-button>\n      </div>\n      <div class=\"svc-sidebar-tabs__separator\">\n        <div></div>\n      </div>\n      <div class=\"svc-sidebar-tabs__items\">\n        <svc-tabs [model]=\"model.topToolbar\"></svc-tabs>\n      </div>\n    </div>\n    <div class=\"svc-sidebar-tabs__bottom-container\">\n      <div class=\"svc-sidebar-tabs__items\">\n        <svc-tabs [model]=\"model.bottomToolbar\"></svc-tabs>\n      </div>\n    </div>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.TabButtonComponent, selector: "svc-tab-button", inputs: ["model"] }, { type: i2.TabsComponent, selector: "svc-tabs", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabControlComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-tab-control",
                    templateUrl: "./tab-control.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-tab-control", TabControlComponent);
